### PR TITLE
ML-143: Clustering results converter

### PR DESCRIPTION
**Pull request by eartser**
Task – [ML-143](https://vyahhi.myjetbrains.com/youtrack/issue/ML-143) Code submission clustering / Prepare for integration

### CODE REVIEW

File: code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt [hunk 1]
```diff
@@ -65,16 +80,41 @@ fun SubmissionsGraph.writeToCsv(outputPath: String) {
...
+fun SubmissionsGraph.writeClustersToTxt(outputPath: String) {
+    val txtFolder = "$outputPath/txt"
+    createFolder(txtFolder)
+    val path = "$txtFolder/clusters.txt"
     val file = File(path)
     file.createNewFile()
     file.writeText(getClusteredGraph().toString())
 }
+
+private fun SubmissionsGraph.toClusteringDataFrame(): DataFrame<*> {
```
Comment (nbirillo) [inline code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt:101]: Please, add tests for it

File: code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt [hunk 2]
```diff
@@ -65,16 +80,41 @@ fun SubmissionsGraph.writeToCsv(outputPath: String) {
...
+    val submissions = mutableListOf<Int>()
+    val clusters = mutableListOf<Int>()
+    val positions = mutableListOf<Int>()
+    getClusteredGraph().graph.vertexSet().forEach { cluster ->
+        var currentPosition = 0
+        cluster.entities.forEach { submissionsNode ->
+            submissionsNode.idList.forEach { submissionId ->
```
Comment (nbirillo) [inline code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt:108]: Should we order solutions by their quality?

File: code-submissions-clustering-plugin/build.gradle.kts [hunk 3]
```diff
@@ -47,18 +49,23 @@ tasks {
             if (project.hasProperty("saveClusters")) {
                 add("--saveClusters")
             }
+            if (project.hasProperty("clusteringResult")) {
```
Comment (nbirillo) [inline code-submissions-clustering-plugin/build.gradle.kts:52]: Please, add examples into the main README file how we can run these tasks and examples of the results

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great!

File: code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt (hunk 2 above)
Comment (eartser) [reply code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt]: We should but cannot do that at the moment as we don't store any information about the quality of code submissions yet. Anyway, fixed assigning positions within cluster here to consider all submissions inside the cluster at a time, not only inside one node.

File: code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt [hunk 4]
```diff
@@ -65,16 +80,39 @@ fun SubmissionsGraph.writeToCsv(outputPath: String) {
...
     file.writeText(getClusteredGraph().toString())
 }
+
+fun SubmissionsGraph.toClusteringDataFrame(): DataFrame<*> {
+    val submissions = mutableListOf<Int>()
+    val clusters = mutableListOf<Int>()
+    val positions = mutableListOf<Int>()
+    getClusteredGraph().graph.vertexSet().forEach { cluster ->
+        var currentPosition = 0
+        cluster.entities.flatMap { it.idList }.sorted().forEach { submissionId ->
```
Comment (nbirillo) [inline code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/LoadUtil.kt:107]: Maybe `forEachIndexed` is better here? Then you can delete currentPosition

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great, only one small comment

**Pull request merged**